shardy/dialect/sdy/ir/attrs.td

```diff
@@ -144,8 +149,8 @@ def Sdy_SubAxisInfo : AttrDef<Sdy_Dialect, "SubAxisInfo"> {
     denoted as follows: `(m)k` for pre-size m and size k.
   }];
   let parameters = (ins
-      "int64_t":$pre_size,
-      "int64_t":$size
+      AttrOrTypeParameter<"int64_t", "pre-size">:$pre_size,
```

Comment:
As discussed offline, lets put something more descriptive than repeating the name of the field, you can take a look at the description of the attr.

Reply:
Please let me know if this is reasonable 😄